src/beamlime/config/builders.py

```diff
@@ -1,145 +1,84 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
-from typing import Literal
-
-from ..resources.templates import (
-    load_app_subscription_tpl,
-    load_application_tpl,
-    load_communication_channel_tpl,
-    load_config_tpl,
-    load_workflow_tpl,
+from functools import partial
+from typing import Any, Callable, Literal, NamedTuple, TypeVar, Union
+
+from ..core.protocols import BeamlimeApplicationProtocol, BeamlimeCommunicationProtocol
+from ..core.system import BeamlimeSystem
+
+_MicroServiceTypes = TypeVar(
+    "_MicroServiceTypes", BeamlimeApplicationProtocol, BeamlimeCommunicationProtocol
 )
 
 
-def _default_application_config(name: str) -> dict:
-    tpl = load_application_tpl()
-    tpl["name"] = name
-    return tpl
+class ServiceRecipe(NamedTuple):
+    name: str
+    constructor: Union[_MicroServiceTypes, str]
+    specs: Any = None
 
 
-def _workflow_config(name: str) -> dict:
-    tpl = load_workflow_tpl()
-    tpl["name"] = name
-    tpl.pop("reference")
-    tpl.pop("process-args")
-    tpl.pop("process-kwargs")
-    return tpl
+def _update_or_pop(config: dict, **kwargs) -> dict:
+    for key, item in kwargs.items():
+        if not item:
+            config.pop(key, None)
+        else:
+            config[key] = item
+    return config
 
 
-def _channel_config(
-    name: str,
-    channel_type: Literal[
-        "MQUEUE", "SQUEUE", "KAFKA-PRODUCER", "KAFKA-CONSUMER", "BULLETIN-BOARD"
-    ] = "SQUEUE",
-    **other_specs,
+def micro_service_config(
```

Comment:
Docstrings would be useful to explain the context, I have no idea what this does, and where it is to be used as part of the larger architecture.